R/LSgameofchance.R

```diff
@@ -147,154 +148,144 @@ compare_function2 <- function(k1, t, p, simulation){
   # add footnote for normalization of prob
-  if(sum(as.numeric(unlist(strsplit(input$p,",")))) != 1){
-    summaryTable$addFootnote("The probability entered does not sum to 1. Already normalized!")
-  }
+  # if(sum(as.numeric(unlist(strsplit(probWin,",")))) != 1){
+  #   summaryTable$addFootnote("The probability entered does not sum to 1. Already normalized!")
+  # }
   
   ## Credible Interval Plot
   CIPlot <- createJaspPlot(title = "Probability of Player 1 Winning",  width = 480, height = 320)
-  CIPlot$dependOn(c("n", "k", "t", "p", "s", "check"))
+  CIPlot$dependOn(c("players", "winPoints", "nSims", "CI"))
   CIPlot$addCitation("JASP Team (2018). JASP (Version 0.9.2) [Computer software].")
   
   # column specification
   CIPlot0 <- ggplot2::ggplot(data= NULL) + 
     #ggplot2::ggtitle("Probability of Player 1 Winning") +
     ggplot2::xlab("Number of Simulated Games") + 
     ggplot2::ylab("Pr(Winning the Game)") +
-    ggplot2::coord_cartesian(xlim = c(0, input$s), ylim = c(0, 1)) 
+    ggplot2::coord_cartesian(xlim = c(0, nSims), ylim = c(0, 1)) 
   
   ## fill in the table and the plot 
-  if (input$n == 2 & max(k) < input$t){
+  if (nPlayers == 2 & max(xPoints) < winPoints){
   
     # output of compare_function1, when there are two players
-    result <- compare_function1(p[1],k[1],k[2],input$t,input$s)
+    result <- compare_function1(probWin[1], xPoints[1], xPoints[2], winPoints, nSims)
     
     # fill in the table
-    summaryTable$addRows(list(players = 1, pPoint = p[1], pA = result[[2]], pS = result[[1]]))
-    summaryTable$addRows(list(players = 2, pPoint = 1-p[1], pA = 1-result[[2]], pS = 1-result[[1]]))
+    summaryTable$addRows(list(players = 1, pPoint = probWin[1],   pointsGained = xPoints[1], pA = result[[2]],   pS = result[[1]]))
+    summaryTable$addRows(list(players = 2, pPoint = 1-probWin[1], pointsGained = xPoints[2], pA = 1-result[[2]], pS = 1-result[[1]]))
     
     # fill in the plot
-    if (input$check){ # whether plot CI or not
+    if (options[["CI"]]){ # whether plot CI or not
       
       # Credibility interval (highest posterior density interval)
       SimulResult <- result[[4]] # store the simulated result
-      SimulMatrix <- matrix(0, nrow = 1000, ncol = input$s) # the matrix of samples from posterior distribution based on simulated result
+      SimulMatrix <- matrix(0, nrow = 1000, ncol = nSims) # the matrix of samples from posterior distribution based on simulated result
       
-      for (i in 1:input$s){  
+      for (i in 1:nSims){  
         SimulMatrix[, i] <- rbeta(1000, SimulResult[i]*i+1, i-SimulResult[i]*i+1)
       }
       CredInt <- apply(SimulMatrix, 2, HDInterval::hdi) # record the credibility interval
       y.upper <- CredInt[1,]
       y.lower <- CredInt[2,]
       CIPlot0 <- CIPlot0 + 
-        ggplot2::geom_polygon(aes(x = c(1:input$s,input$s:1), y = c(y.upper, rev(y.lower))), 
+        ggplot2::geom_polygon(ggplot2::aes(x = c(1:nSims,nSims:1), y = c(y.upper, rev(y.lower))), 
```

Comment:
all non-base functions should be called with the package prefix, including the `aes` call from `ggplot2` from example